pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -168,37 +174,51 @@ public void removeSegmentsFromStore(String tableNameWithType, List<String> segme
 
   protected void removeSegmentFromStore(String tableNameWithType, String segmentId) {
     final String rawTableName = TableNameBuilder.extractRawTableName(tableNameWithType);
-    if (_localDiskDir != null) {
-      File fileToMove = new File(new File(_localDiskDir, rawTableName), segmentId);
-      if (fileToMove.exists()) {
-        File targetDir = new File(new File(_localDiskDir, DELETED_SEGMENTS), rawTableName);
-        try {
+    if (_dataDir != null) {
+      URI deletedSegmentDirURI;
+      URI fileToMoveURI;
+      PinotFS pinotFS;
+      try {
+        URI dataDirURI = ControllerConf.getUriFromPath(_dataDir);
+        fileToMoveURI = ControllerConf.constructSegmentLocation(_dataDir, rawTableName, segmentId);
+        pinotFS = PinotFSFactory.create(dataDirURI.getScheme());
+        deletedSegmentDirURI = ControllerConf.getUriFromPath(StringUtil.join(File.separator, _dataDir, DELETED_SEGMENTS, rawTableName));
+      } catch (URISyntaxException e) {
+        LOGGER.error("Could not create DELETED_SEGMENTS uri with dataDir {}, tableName {}", _dataDir, rawTableName);
+        throw new RuntimeException(e);
+      }
+
+      try {
+        if (pinotFS.exists(fileToMoveURI)) {
           // Overwrites the file if it already exists in the target directory.
-          FileUtils.copyFileToDirectory(fileToMove, targetDir, false);
-          LOGGER.info("Moved segment {} from {} to {}", segmentId, fileToMove.getAbsolutePath(), targetDir.getAbsolutePath());
-          if (!fileToMove.delete()) {
-            LOGGER.warn("Could not delete file", segmentId, fileToMove.getAbsolutePath());
+          pinotFS.move(fileToMoveURI, deletedSegmentDirURI);
+          LOGGER.info("Moved segment {} from {} to {}", segmentId, fileToMoveURI.toString(), deletedSegmentDirURI.toString());
+          if (!pinotFS.delete(fileToMoveURI)) {
```

Comment:
since you are calling pinotfs.move, you should not be deleting the old file here. you will always get an error. just make sure that the pinotfs.move does not throw an error if there is already a destination file